python/pynessie/cli.py

```diff
@@ -333,12 +335,13 @@ def branch_(
 @click.option("-l", "--list", cls=MutuallyExclusiveOption, is_flag=True, help="list branches", mutually_exclusive=["delete"])
 @click.option("-d", "--delete", cls=MutuallyExclusiveOption, is_flag=True, help="delete a branches", mutually_exclusive=["list"])
 @click.option("-f", "--force", is_flag=True, help="force branch assignment")
+@click.option("-o", "--hash-on-ref", help="Hash on source-reference")
```

Comment:
could you elaborate why we need `--hash-on-ref`? It's not immediately obvious

Reply:
added some more docs about that